lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -5463,6 +5463,13 @@ private void tooManyDocs(long addedNumDocs) {
     throw new IllegalArgumentException("number of documents in the index cannot exceed " + actualMaxDocs + " (current document count is " + pendingNumDocs.get() + "; added numDocs is " + addedNumDocs + ")");
   }
 
+  /**
+   * Returns the number of documents in the index including documents are being added (i.e., reserved).
+   */
```

Comment:
Maybe add `@lucene.experimental`?  We are exposing (slightly) internal details about `IndexWriter` so maybe we need to reserve the right to change this API in the future ...

Reply:
++. I pushed https://github.com/apache/lucene-solr/pull/1941/commits/61d4fbc4a01cbfc50f386a8a88064e82d1027ee0